src/test/javascript/spec/component/statistics/statistics-graph.component.spec.ts

```diff
@@ -0,0 +1,125 @@
+                { provide: TranslateService, useClass: MockTranslateService },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(StatisticsGraphComponent);
+                component = fixture.componentInstance;
+                service = TestBed.inject(StatisticsService);
+                httpMock = TestBed.get(HttpTestingController);
+            });
+    }));
+
+    afterEach(fakeAsync(() => {
+        jest.clearAllMocks();
+    }));
+
+    it('should initialize', fakeAsync(() => {
+        let graphData: number[] = [];
+        component.graphType = Graphs.SUBMISSIONS;
+        let arrayLength = 0;
+        const spy = spyOn(service, 'getChartData');
+
+        for (const span of Object.values(SpanType)) {
+            component.currentSpan = span;
+            switch (span) {
+                case SpanType.DAY:
+                    arrayLength = 24;
+                    break;
+                case SpanType.WEEK:
+                    arrayLength = 7;
+                    break;
+                case SpanType.MONTH:
+                    const startDate = moment().subtract(1, 'months');
+                    arrayLength = moment().diff(startDate, 'days');
+                    break;
+                case SpanType.QUARTER:
+                    arrayLength = 12;
+                    break;
+                case SpanType.YEAR:
+                    arrayLength = 12;
+                    break;
+            }
+            graphData = [];
+            for (let i = 0; i < arrayLength; i++) {
+                graphData[i] = i + 1;
+            }
+            spy.and.returnValue(of(graphData));
+
+            component.ngOnInit();
+            tick();
+
+            expect(component.dataForSpanType).to.equal(graphData);
+            expect(component.chartData[0].data).to.equal(graphData);
+        }
+    }));
+
+    it('should initialize after changes', fakeAsync(() => {
+        component.graphType = Graphs.SUBMISSIONS;
+        component.currentSpan = SpanType.WEEK;
+        const changes = { currentSpan: { currentValue: SpanType.DAY } as SimpleChange };
+        const graphData = [];
+        for (let i = 0; i < 24; i++) {
+            graphData[i] = i + 1;
+        }
+
+        component.ngOnChanges(changes);
+        const req = httpMock.expectOne({ method: 'GET' });
+        req.flush([...graphData]);
+        tick();
+
+        expect(component.dataForSpanType).to.deep.equal(graphData);
+        expect(component.currentSpan).to.equal(SpanType.DAY);
+    }));
+
+    it('should switch time span', fakeAsync(() => {
+        component.graphType = Graphs.SUBMISSIONS;
+        component.currentSpan = SpanType.WEEK;
+        const graphData = [1, 2, 3, 4, 5, 6, 8];
+        spyOn(service, 'getChartData').and.returnValue(of(graphData));
+
+        fixture.detectChanges();
+        tick();
```

Comment:
Does the test fail without ticking? I wouldn't expect this to be needed.

Reply:
Maybe i could need a little bit of elaboration about tick() :P I thought that tick() is needed when the call which is tested includes an Observable, because tick() then triggered the result and gives it to the Subscriber. However i removed all the tick() in my test cases and it still works now.

Reply:
I think that is because the spy is returning a value, so we don't strictly need to wait asynchronously.